### PR TITLE
Report whether k6 runs on CI

### LINKED ISSUE #4038: Report whether a test was run from a CI environment in usage report data
### Feature Description

k6 collects anonymized usage data with users content. In order to help us continue improving k6, we would like to add the information whether a test was run in a CI environment to such report.

### Suggested Solution (optional)

The currently proposed solution is to look for the `CI` variable in the system's environment, and if set, update the usage report data accordingly (as proposed by @joanlopez). 

### Already existing or connected issues / PRs (optional)

_No response_

### PR DESCRIPTION
**Pull request by joanlopez**
## What?

It adds a new field into the usage report that tells whether the `CI` environment variable is defined or not when k6 is executed.

## Why?

Because as part of understanding how users use k6, we want to know how many of total k6 executions happen in a CI system, and although it's probably difficult to have a complete picture, this is the first naive approach to start collecting that data, as the `CI` environment variable is some sort of non-written standard, that most of the CI systems (and other tools) use, and set.

## Checklist

- [X] I have performed a self-review of my code.
- [X] I have added tests for my changes.
- [X] I have run linter locally (`make lint`) and all checks pass.
- [X] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

Closes #4038

### CODE REVIEW

**Comment by inancgumus** [comment]
Removing myself from the reviewers as we pair-coded this :)

File: cmd/report.go [hunk 1]
```diff
@@ -28,6 +28,11 @@ func createReport(u *usage.Usage, execScheduler *execution.Scheduler) map[string
 	}
 	m["executors"] = executors
 
+	_, isCI := execScheduler.GetState().Test.LookupEnv("CI")
```
Comment (olegbespalov) [inline cmd/report.go:31]:
```suggestion
	_, isCI := execState.Test.LookupEnv("CI")
```

File: cmd/report_test.go [hunk 2]
```diff
@@ -46,4 +49,5 @@ func TestCreateReport(t *testing.T) {
 	assert.EqualValues(t, 6, m["vus_max"])
 	assert.EqualValues(t, 170, m["iterations"])
 	assert.NotEqual(t, "0s", m["duration"])
+	assert.EqualValues(t, map[string]any{"ci": true}, m["env"])
```
Comment (olegbespalov) [inline cmd/report_test.go:52]: I'd rather suggest testing that by default the CI's value is `false` and introduce a dedicated test case, where we can test that in the CI environment we do reporting that it sets.

**Review by olegbespalov — COMMENTED** [review]
The approach with the CI looks good to me :+1: Left a few non-blocking suggestions.

Quickly [searching on GitHub](https://github.com/watson/ci-info/blob/master/index.js#L35-L48) (or I believe there could be other popular packages) seems like there could be a couple of more environment variables that we can look into. 

Including such environment variables still sounds pretty safe, but our statistics will be more accurate, and unlikely we return to adjusting this code soon, so it's good to consider them since the beginning.

File: cmd/report.go (hunk 1 above)
Comment (olegbespalov) [reply cmd/report.go:31]: And it's probably still better to check the case `CI=false`

**Comment by joanlopez** [comment]
> Quickly [searching on GitHub](https://github.com/watson/ci-info/blob/master/index.js#L35-L48) (or I believe there could be other popular packages) seems like there could be a couple of more environment variables that we can look into.
> 
> Including such environment variables still sounds pretty safe, but our statistics will be more accurate, and unlikely we return to adjusting this code soon, so it's good to consider them since the beginning.

> And it's probably still better to check the case `CI=false`

Considering these two comments, what do you think about something like the following:

```go
// isCI is a helper that follows a naive approach to determine if k6 is being
// executed within a CI system. This naive approach consists of checking if
// the "CI" environment variable (among others) is set.
//
// We treat the "CI" environment variable carefully, because it's the one
// used more often, and because we know sometimes it's explicitly set to
// "false" to signal that k6 is not running in a CI environment.
//
// It is not a foolproof method, but it should work for most cases.
func isCI(lookupEnv func(key string) (val string, ok bool)) bool {
	if ci, ok := lookupEnv("CI"); ok {
		ciBool, err := strconv.ParseBool(ci)
		if err == nil {
			return ciBool
		}
		// If we can't parse the "CI" value as a bool, we assume return true
		// because we know that at least it's not set to any variant of "false",
		// which is the most common use case, and the reasoning we apply below.
		return true
	}

	// List of common environment variables used by different CI systems.
	ciEnvVars := []string{
		"BUILD_ID",               // Jenkins, Cloudbees
		"BUILD_NUMBER",           // Jenkins, TeamCity
		"CI",                     // Travis CI, CircleCI, Cirrus CI, Gitlab CI, Appveyor, CodeShip, dsari
		"CI_APP_ID",              // Appflow
		"CI_BUILD_ID",            // Appflow
		"CI_BUILD_NUMBER",        // Appflow
		"CI_NAME",                // Codeship and others
		"CONTINUOUS_INTEGRATION", // Travis CI, Cirrus CI
		"RUN_ID",                 // TaskCluster, dsari
	}

	// Check if any of these variables are set
	for _, key := range ciEnvVars {
		if _, ok := lookupEnv(key); ok {
			return true
		}
	}

	return false
}
```

cc/ @inancgumus

**Review by olegbespalov — APPROVED** [review]
:rocket:

**Review by mstoykov — APPROVED** [review]
LGTM, but IMO the current name name in the report is not accurate

File: cmd/report.go [hunk 3]
```diff
@@ -27,6 +28,9 @@ func createReport(u *usage.Usage, execScheduler *execution.Scheduler) map[string
 		executors[ec.GetType()]++
 	}
 	m["executors"] = executors
+	m["env"] = map[string]any{
+		"ci": isCI(execState.Test.LookupEnv),
```
[the remark this answers is not in the document]
Comment (joanlopez) [reply cmd/report.go:32]: Yeah! I kinda agree, although to be honest when we thought about `env`, it wasn't strictly as a direct correlation with environment variables but as an umbrella for anything related with the environment.

Although, arguably, OS, arch, etc are also values associated with the environment where it runs.

No strong opinion, tbh. What's your suggestion?
cc/ @olegbespalov @inancgumus

[the remark this answers is not in the document]
Comment (mstoykov) [reply cmd/report.go:32]: > What's your suggestion?

Move it up a level as all the other ones you listed :)

[the remark this answers is not in the document]
Comment (olegbespalov) [reply cmd/report.go:32]: I don't have a strong opinion, slightly not sure if we ever need anything else in `env`, keeping in mind privacy reasons and closed list number of the things that we collect. Moreover, like @joanlopez mentioned other values associated with the environment, so probably we could keep new property on the same level as @mstoykov  suggesting.

[the remark this answers is not in the document]
Comment (inancgumus) [reply cmd/report.go:32]: No strong opinion. `ci` at the top-level is also fine.

**Review by inancgumus — APPROVED** [review]
Nice work 👍

File: cmd/report.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (joanlopez) [reply cmd/report.go]: Renamed and moved to the top level, as suggested 👍🏻

File: cmd/report_test.go [hunk 4]
```diff
@@ -18,32 +18,90 @@ import (
...
+
+		m := createReport(usage.New(), s)
+		require.NoError(t, err)
+
+		assert.Equal(t, consts.Version, m["k6_version"])
+		assert.EqualValues(t, map[string]int{"shared-iterations": 1}, m["executors"])
+		assert.EqualValues(t, 6, m["vus_max"])
+		assert.EqualValues(t, 170, m["iterations"])
+		assert.NotEqual(t, "0s", m["duration"])
+		assert.EqualValues(t, map[string]any{"ci": false}, m["env"])
```
Comment (olegbespalov) [inline cmd/report_test.go:61]: Here and other tests :+1: 
```suggestion
		assert.EqualValues(t, false, m["is_ci"])
```

Comment (joanlopez) [reply cmd/report_test.go:61]: Yeah, yeah, sorry! Forgot to add the changes on the test file, pushing now! Apologies for the noise! 🙏🏻

**Review by olegbespalov — APPROVED** [review]

**Review by inancgumus — APPROVED** [review]

**Pull request merged**